### ISSUE TITLE
[monitoring] The Integration Activity Table should allow for checking detailed activity logs

### ISSUE THREAD

**Issue by deeleman** [syndesisio/syndesis#1855]
## This is a...

<pre><code>
[x] Feature request
[ ] Regression (a behavior that used to work and stopped working in a new release)
[ ] Bug report  
[ ] Documentation issue or request
</code></pre>

## The problem
Currently, the activity tab in the integration detail view shows a non-detailed listview of activity entries. Such information is pretty basic and we need to broaden each activity details upon request.

## Expected behavior
By clicking on each row, a nested view will unfold displaying a table with a corresponding detailview featuring the following:

- Step name
- Time (isn't this redundant, being more likely the same for all the items displayed at the table?)
- Duration (in seconds)
- Status (`Error` | `Success`)
- Output (plain text or `n/a` when blank)

## Screenshot
<img width="915" alt="captura de pantalla 2018-03-06 a las 20 30 03" src="https://user-images.githubusercontent.com/1104146/37054203-7f873d64-217e-11e8-84cb-ef2ded30fa42.png">

## API Endpoints and Schemas

TBD @chirino Could you fill this out? Thx

## Tasks involved / Steps to Reproduce

1. Create and deploy an integration. Wait for its `published` status.
2. Go to the integration detail view and click on activity
3. On the activity rows displayed (if any), click on the arrow on the left.
4. Check that a sub view contained tabular data matching the design above is rendered

**Comment by deeleman** [comment]
This is an extracted item from the activity logs graph:

```json
{
  "id": "i-L75XqcoCtaqO-wCSP90z",
  "logts": "2018-03-08T16:57:02.775513749Z",
  "at": 1520528222772,
  "pod": "i-my-db-to-db-integration-2-rvrxs",
  "ver": "1",
  "status": "done",
  "failed": false,
  "steps": [
    {
      "id": "i-L75XqcoCtaqO-wCSP91z",
      "at": 1520528222772,
      "duration": 603998
    },
    {
      "id": "i-L75XqcpCtaqO-wCSP92z",
      "at": 1520528222773,
      "duration": 6304147
    }
  ]
}
```

As you can see in this real example, the `steps` array property contains two elements, that should populate the detailed activity logs subview per log item. Unfortunately, the design provided requires us to implement the following:

- Step name: ✗
- Time: ✔
- Duration: ✔
- Status: ✗
- Output: **???**

I've tried to create get hold of the step name at least by creating a very complex projection by traversing the integration graph to get its steps and the activity log item steps, but it is not possible, since there is no way to pair the steps enlisted in one graph with this other one. Even if it was possible, the resulting projection would be quite cumbersome. A serverside solution would seem more feasible IMO.

In a nutshell, and in order to complete this task, we need the graph posted above to contain the missing information for each step object: `name`, `status`and `output`(whereas required).

**Comment by chirino** [comment]
Step name is defined in the integration model. Use the step id to look it up. Unless failure is set, status is “ok”.  Output would be the failure message or if messages field is set. See the model defined in ActivityStep for actual field names.

**Comment by deeleman** [comment]
> Step name is defined in the integration model. Use the step id to look it up. 

Please take a minute to review my previous message again: I tried to project the required steps information from the integration model already, but the step IDs there and the steps ID in the activity logs do not match. There is no way whatsoever to correlate one graph with the other. 

Eg: This is an excerpt of my integration, focusing on the steps defined:

```json
{
  "steps": [
    {
      "connection": {
        "icon": "fa-database",
        "connectorId": "sql",
        "name": "PostgresDB",
        "id": "5"
      },
      "action": {
        "name": "Periodic SQL invocation"
      },
      "kind": "step",
      "stepKind": "endpoint",
      "id": "-L757zb8y-dUJU4XL3Gz"
    },
    {
      "action": {},
      "kind": "step",
      "stepKind": "mapper",
      "name": "Data Mapper",
      "id": "-L7584USy-dUJU4XL3H-"
    },
    {
      "connection": {
        "icon": "fa-database",
        "connectorId": "sql",
        "name": "PostgresDB",
        "id": "5"
      },
      "action": {
        "name": "Invoke stored procedure"
      },
      "kind": "step",
      "stepKind": "endpoint",
      "id": "-L7583CMy-dUJU4XL3Gz"
    }
  ]
}
```

And here we have a snapshot of the last 9 activity log entries (BTW the recordset returned by the API is capped to 9 deltas - I was unaware of this and instead of persisting all the records by appending the newer ones to the store, the last 9 records fetched squash the pre-existing items - this will have to be tweaked at some point otherwise this is not a relevant log, thanks we have Redux in place for that).

Pick any step ID of your choice from the previous graph and try find a match below: There's none.

```json
[
  {
    "id": "i-L75Xt46CtaqO-wCSPB2z",
    "logts": "2018-03-08T16:57:12.778431387Z",
    "at": 1520528232775,
    "pod": "i-my-db-to-db-integration-2-rvrxs",
    "ver": "1",
    "status": "done",
    "failed": false,
    "steps": [
      {
        "id": "i-L75Xt46CtaqO-wCSPB3z",
        "at": 1520528232775,
        "duration": 963418
      }
    ]
  },
  {
    "id": "i-L75XspUCtaqO-wCSPAqz",
    "logts": "2018-03-08T16:57:11.791582158Z",
    "at": 1520528231775,
    "pod": "i-my-db-to-db-integration-2-rvrxs",
    "ver": "1",
    "status": "done",
    "failed": false,
    "steps": [
      {
        "id": "i-L75XspWCtaqO-wCSPAsz",
        "at": 1520528231777,
        "duration": 14090352
      }
    ]
  },
  {
    "id": "i-L75Xs_rCtaqO-wCSPAdz",
    "logts": "2018-03-08T16:57:10.778937734Z",
    "at": 1520528230775,
    "pod": "i-my-db-to-db-integration-2-rvrxs",
    "ver": "1",
    "status": "done",
    "failed": false,
    "steps": [
      {
        "id": "i-L75Xs_sCtaqO-wCSPAez",
        "at": 1520528230776,
        "duration": 1178611
      }
    ]
  },
  {
    "id": "i-L75XsLECtaqO-wCSPARz",
    "logts": "2018-03-08T16:57:09.777938684Z",
    "at": 1520528229775,
    "pod": "i-my-db-to-db-integration-2-rvrxs",
    "ver": "1",
    "status": "done",
    "failed": false,
    "steps": [
      {
        "id": "i-L75XsLECtaqO-wCSPASz",
        "at": 1520528229775,
        "duration": 703948
      },
      {
        "id": "i-L75XsLFCtaqO-wCSPATz",
        "at": 1520528229776,
        "duration": 14382005
      }
    ]
  },
  {
    "id": "i-L75Xs5bCtaqO-wCSPAEz",
    "logts": "2018-03-08T16:57:08.779212297Z",
    "at": 1520528228775,
    "pod": "i-my-db-to-db-integration-2-rvrxs",
    "ver": "1",
    "status": "done",
    "failed": false,
    "steps": [
      {
        "id": "i-L75Xs5bCtaqO-wCSPAFz",
        "at": 1520528228775,
        "duration": 1022888
      },
      {
        "id": "i-L75Xs5cCtaqO-wCSPAGz",
        "at": 1520528228776,
        "duration": 16075958
      }
    ]
  },
  {
    "id": "i-L75XrqzCtaqO-wCSPA1z",
    "logts": "2018-03-08T16:57:07.776439634Z",
    "at": 1520528227775,
    "pod": "i-my-db-to-db-integration-2-rvrxs",
    "ver": "1",
    "status": "done",
    "failed": false,
    "steps": [
      {
        "id": "i-L75Xrr-CtaqO-wCSPA2z",
        "at": 1520528227776,
        "duration": 1260770
      },
      {
        "id": "i-L75Xrr0CtaqO-wCSPA3z",
        "at": 1520528227777,
        "duration": 10629582
      }
    ]
  },
  {
    "id": "i-L75XrbMCtaqO-wCSP9pz",
    "logts": "2018-03-08T16:57:06.775515947Z",
    "at": 1520528226775,
    "pod": "i-my-db-to-db-integration-2-rvrxs",
    "ver": "1",
    "status": "done",
    "failed": false,
    "steps": [
      {
        "id": "i-L75XrbMCtaqO-wCSP9qz",
        "at": 1520528226775,
        "duration": 1399900
      }
    ]
  },
  {
    "id": "i-L75XrMiCtaqO-wCSP9cz",
    "logts": "2018-03-08T16:57:05.776617346Z",
    "at": 1520528225774,
    "pod": "i-my-db-to-db-integration-2-rvrxs",
    "ver": "1",
    "status": "done",
    "failed": false,
    "steps": [
      {
        "id": "i-L75XrMiCtaqO-wCSP9dz",
        "at": 1520528225774,
        "duration": 4073862
      },
      {
        "id": "i-L75XrMmCtaqO-wCSP9ez",
        "at": 1520528225778,
        "duration": 9219588
      }
    ]
  },
  {
    "id": "i-L75Xr74CtaqO-wCSP9Qz",
    "logts": "2018-03-08T16:57:04.774096257Z",
    "at": 1520528224773,
    "pod": "i-my-db-to-db-integration-2-rvrxs",
    "ver": "1",
    "status": "done",
    "failed": false,
    "steps": [
      {
        "id": "i-L75Xr74CtaqO-wCSP9Rz",
        "at": 1520528224773,
        "duration": 3027286
      },
      {
        "id": "i-L75Xr77CtaqO-wCSP9Sz",
        "at": 1520528224776,
        "duration": 18822084
      }
    ]
  },
  {
    "id": "i-L75XqcoCtaqO-wCSP90z",
    "logts": "2018-03-08T16:57:02.775513749Z",
    "at": 1520528222772,
    "pod": "i-my-db-to-db-integration-2-rvrxs",
    "ver": "1",
    "status": "done",
    "failed": false,
    "steps": [
      {
        "id": "i-L75XqcoCtaqO-wCSP91z",
        "at": 1520528222772,
        "duration": 603998
      },
      {
        "id": "i-L75XqcpCtaqO-wCSP92z",
        "at": 1520528222773,
        "duration": 6304147
      }
    ]
  }
]
```

**Comment by chirino** [comment]
Ok. That’s a bug then. We need the step Id there for you to correlate.

**Comment by chirino** [comment]
@deeleman once https://github.com/syndesisio/syndesis/pull/1922 merges, you should be able to do the correlation.

**Comment by deeleman** [comment]
Yay! I just saw this now. Will be keeping an eye on it.

**Comment by deeleman** [comment]
The step ids found in the activity records still do not match the ones in the integration overview graph

**Comment by deeleman** [comment]
As you can see by the screenshot attached below, the current state of the UI DOES NOT match the screenshot or design provided. This is because the Patternfly components feature a complete different look and feel than the design provided. 

![captura de pantalla 2018-03-12 a las 14 46 45](https://user-images.githubusercontent.com/1104146/37287195-87a7a74c-2604-11e8-90fd-7a79c7c19fc2.png)


There're 2 ways to make our current implementation matching the designs provided:

- Have Patternfly tweaking their product (which is unlikely)
- Have UI (me) developing a totally custom UI for this (which entails "hacking" the current implementation of PF for our project the dirty way rather than generating new components, actually)

Another workaround is to leave this as is, bearing in mind that this is how the PF components are meant to be rendered, and just tweak margins and paddings here and there.

//cc @syndesisio/uxd @sjcox-rh @amysueg

**Comment by michael-coker** [comment]
@deeleman Using this [list view expansion](https://rawgit.com/patternfly/patternfly/master-dist/dist/tests/list-view-simple-expansion.html) should give you the border and drop shadow around the expanded row. The rest of the existing layout looks OK to me.

**Comment by amysueg** [comment]
Thank you, @michael-coker. I was just writing a reply when I saw yours.

@deeleman - It seems that the last column "Action" is missing, or is that planned for a later sprint?

**Comment by deeleman** [comment]
> It seems that the last column "Action" is missing

@amysueg Nothing is missing here, Amy. It's just that the granularity of the issues has been increased to make them actionable in an agile way. Otherwise there's no way to really know what needs to get done and properly track down the contribution of each party involved.

The feature you refer to is defined in #1856 

Back tot he topi, aside from the border and the drop shadow, are you guys okay with the table layout and the current margin/padding setup?

**Comment by michael-coker** [comment]
@deeleman I don't think there is a patternfly-ng component for that list view. Can we use an ng-template with the markup in the [list view expansion](https://rawgit.com/patternfly/patternfly/master-dist/dist/tests/list-view-simple-expansion.html) demo instead?

**Comment by deeleman** [comment]
> I don't think there is a patternfly-ng component for that list view. Can we use an ng-template with the markup in the list view expansion demo instead?

No, I'm afraid that component styling does not operate that way. In any event [this PR](https://github.com/syndesisio/syndesis/pull/1962) provides an alternative solution.

**Comment by chirino** [comment]
@deeleman @dongniwang I agree the timestamp per step seems redundant.  For fast executions it's typically the same timestamp for all steps.  For slow ones, you can sort of deduce it by adding up the duration of each step.  It takes up alot of space in the table.  Could we please remove it?

**Comment by chirino** [comment]
Once PR #2038 goes in I think this issue is closable.
